src/domain/transaction/evm/evm_transaction.rs

```diff
@@ -272,6 +296,17 @@ where
         let updated_evm_data =
             updated_evm_data.with_signed_transaction_data(sig_result.into_evm()?);
 
+        let total_cost = self.calculate_total_cost(&tx).await?;
+
+        EvmTransactionValidator::validate_sufficient_relayer_balance(
```

Comment:
In addition to throwing error we need to make sure that tx is not retried for sending any more. We should change TX status to Failed and ensure that webhook notification is sent.

I think it also makes sense to extend tx repo model to add some failure reason field and use it in tx response to be more transparent so users could take actions based on failure reason(fund + resend for example).

Reply:
Yes
if somebody tries to get tx by id, or when user receives webhook that tx has failed best to include filed that denotes why it failed